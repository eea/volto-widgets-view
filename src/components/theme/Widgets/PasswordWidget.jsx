import React from 'react';
import cx from 'classnames';

export const PasswordWidget = ({ children, className }) => (
  <span className={cx(className, 'password', 'widget')}>
    {children ? '********' : ''}
  </span>
);
