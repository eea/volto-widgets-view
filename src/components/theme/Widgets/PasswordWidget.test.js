import React from 'react';
import renderer from 'react-test-renderer';
import { PasswordWidget } from './PasswordWidget';

describe('PasswordWidget', () => {
  it('renders an empty password view widget component', () => {
    const component = renderer.create(<PasswordWidget />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a password view widget component', () => {
    const component = renderer.create(
      <PasswordWidget className="metadata">
        A very strong password
      </PasswordWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
