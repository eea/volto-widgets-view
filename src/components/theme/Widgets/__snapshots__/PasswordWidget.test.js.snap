// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordWidget renders a password view widget component 1`] = `
<span
  className="metadata password widget"
>
  ********
</span>
`;

exports[`PasswordWidget renders an empty password view widget component 1`] = `
<span
  className="password widget"
>
  
</span>
`;
